apps/website/docs/learn/02-environment-setup.md

```diff
@@ -147,8 +150,174 @@ export function App() {
+
+## Babel configuration
+
+When using Next.js App Router, babel is not the default compiler, but can still be used.
+Create or modify a `babelLoader.config.js` file as follows (note that it's not a `babel.config.js`). This is used to optimize builds and enables static extraction of CSS for web. Learn how to configure the [babel-preset](/api/babel-preset/) in the API docs.
+
+```js title="babelLoader.config.js"
+import path from "path";
+
+const config = {
+  presets: [
+    [
+      "next/dist/compiled/babel/preset-typescript",
+      {
+        allowNamespaces: true,
+      },
+    ],
+    [
+      "react-strict-dom/babel-preset",
+      {
+        debug: true,
+        dev: process.env.NODE_ENV === "development",
+        rootDir: process.cwd(),
+        // if you want to use @/* from tsconfig
+        // a patch is required in node_modules/react-strict-dom/babel/preset.js
+        // see https://github.com/facebook/react-strict-dom/pull/294 for more information
+        // aliases: {
+        //   "@/*": [path.join(process.cwd(), "*")],
+        // },
+      },
+    ],
+  ],
+};
+
+export default config;
+
+```
+
+## PostCSS configuration
+
+[PostCSS](https://postcss.org/) is a tool for generating CSS. It's enabled by default in Next.js and it's the recommended way to extract React Strict DOM styles to static CSS for web builds. `react-strict-dom/postcss-plugin` can be used to extract styles. Create a `postcss.config.mjs` file as follows.
+
+```js title="postcss.config.mjs"
+// note that you need to import manually the babel loader so configuration is shared between next.js & postcss compilation
+import babelLoader from "./babelLoader.config.js";
+
+const config = {
+  plugins: {
+    "react-strict-dom/postcss-plugin": {
+      include: [
+        // Include source files to watch for style changes
+        'src/**/*.{js,jsx,mjs,ts,tsx}',
+        // List any installed node_modules that include UI built with React Strict DOM
+        'node_modules/<package-name>/*.js'
+      ],
+      babelConfig: babelLoader,
+      useLayers: true,
+    },
+    autoprefixer: {},
+  },
+};
+
+export default config;
+```
+
+## Next.js configuration
+
+Create or edit the `next.config.js` file as follows. Note that below you will find config for both turbopack or webpack.
+
+```js title="next.config.js"
+import type { NextConfig } from "next";
+
+import babelLoader from "./babelLoader.config.js";
+
+function getBabelLoader() {
+  return {
+    loader: "babel-loader",
+    options: babelLoader,
+  };
+}
+
+const nextConfig: NextConfig = {
+  transpilePackages: ["react-strict-dom"],
+
+  turbopack: {
+    rules: {
+      "*.{js,jsx,ts,tsx}": {
+        loaders: [getBabelLoader()],
+      },
+    },
+  },
+
+  webpack: (config, { webpack }) => {
+    config.resolve.mainFields = ["module", "main"];
+
+    config.module.rules.push({
```

Comment:
We should add `exclude` option in the module rule so that unrelated node_modules should not go through this babel loader. 
```typescript
        exclude: /node_modules(?!\/react-strict-dom)/,
```

See:
https://github.com/facebook/stylex/blob/4a4fe98486cf99a9783e10ec07a41452790a17a6/examples/example-nextjs/next.config.js#L21